### DISCUSSION TITLE
CSP not applied to all pages

### DISCUSSION THREAD

**Discussion by justincavery** [vercel/vercel#6611] [Help]
I'm trying to apply CSP headers to all the pages on our site (NextJS).

Initially I have the following rules 

```
  async headers() {
    return [
      {
        source: '/:path*',
        headers: [
          {
            key: 'X-Frame-Options',
            value: 'DENY',
          },
          {
            key: 'X-XSS-Protection',
            value: '1; mode=block',
          },
          {
            key: 'Content-Security-Policy',
            value: cspRules,
          },
        ],
      },
    ];
  },
};
```

This would apply CSP headers to the main domain, but landing on any pages within the site did _not_ have the CSP applied.

We updated the rule from `source: '/:path*',` to be `source: '/(.*)',` and this then made every page within the site include the CSP, but the homepage no longer contained it.

Finally, I added two blocks for each source, but that resulted in only the subpages having the CSP applied.

How do I ensure the domain has the CSP applied?

**Comment by leerob** [comment]
> Finally, I added two blocks for each source, but that resulted in only the subpages having the CSP applied.

I'm not following this part - so it was working, then you added a different rule for the home page specifically, but it didn't apply?

**Reply by justincavery** [reply]
The final approach was to include both sets of source with the same rules

```
async headers() {
    return [
      {
        source: '/(.*)',
        headers: [
          {
            key: 'X-Frame-Options',
            value: 'DENY',
          },
          {
            key: 'X-XSS-Protection',
            value: '1; mode=block',
          },
          {
            key: 'Content-Security-Policy',
            value: cspRules,
          },
        ],
      },
      {
        source: '/',
        headers: [
          {
            key: 'X-Frame-Options',
            value: 'DENY',
          },
          {
            key: 'X-XSS-Protection',
            value: '1; mode=block',
          },
          {
            key: 'Content-Security-Policy',
            value: cspRules,
          },
        ],
      },
    ];
  }
```

In this case I was _hoping_ that the first block would apply the CSP to the sub pages, and the second block would apply the rule to the homepage (which isn't getting the rule applied when I just use `source: '/(.*)'`

**Comment by justincavery** [comment]
@leerob does this clear up the issue we're experiencing?
